app/models/submission.rb

```diff
@@ -105,13 +110,25 @@ def filename_for_bulk_download
    		.gsub(/[^0-9A-Za-z.\-]/, '_')
   end
 
+  def filename_for_corrections_bulk_download
+		(team.first(180) + '-' +
+			last_modification_by_users_at.strftime("%F-%H%M") +
+			(too_late? ? '-LATE' : '') +
+			+ '-ID-' + id + '-CORRECTION' +
+			assignment.accepted_file_type)
+			.gsub(/[\x00\/\\:\*\?\"<>\|]/, '_')
+	   	.gsub(/^.*(\\|\/)/, '')
+   		# Strip out the non-ascii characters
+   		.gsub(/[^0-9A-Za-z.\-]/, '_')
+  end
+
```

Comment:
It seems to me that `filename_for_corrections_bulk_download` and `filename_for_submissions_bulk_download` are quite similar, so it should be possible to refactor this (e.g. using some common private method that they both use) in order to avoid code duplication.

Reply:
done. I also moved the `filename_for_bulk_download` to `private`.